### PR TITLE
Fix sanity check for RichProgressBar

### LINKED ISSUE #1: Proposal for help
Hi @williamFalcon  ! I saw your project and I am very pleased by the idea. I wish to help you writing production level code.  PLease let me know in what way can I help!

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Fixes sanity check for RichProgressBar

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

Fixes #1 

cc @borda @tchaton @SeanNaren @kaushikb11

### CODE REVIEW

File: tests/callbacks/test_rich_progress_bar.py [hunk 1]
```diff
@@ -201,3 +201,24 @@ def test_rich_progress_bar_refresh_rate(progress_update, tmpdir, refresh_rate, e
     trainer.fit(model)
 
     assert progress_update.call_count == expected_call_count
+
+
+@RunIf(rich=True)
+def test_rich_progress_bar_stages(tmpdir):
```
Comment (awaelchli) [inline tests/callbacks/test_rich_progress_bar.py:207]: nothing is being tested here?

**Review by awaelchli — APPROVED** [review]
I would drop the test

File: tests/callbacks/test_rich_progress_bar.py (hunk 1 above)
Comment (kaushikb11) [reply tests/callbacks/test_rich_progress_bar.py:207]: We missed testing for sanity check steps, hence we didn't catch the bug.

Comment (awaelchli) [reply tests/callbacks/test_rich_progress_bar.py:207]: Here we have two tests for the sanity check stage for an example: https://github.com/PyTorchLightning/pytorch-lightning/blob/ff20af41fd363f9520f5e0083d7a49051353cd1c/tests/callbacks/test_tqdm_progress_bar.py#L166

Comment (kaushikb11) [reply tests/callbacks/test_rich_progress_bar.py]: Test updated!

**Review by rohitgr7 — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

**Review by ananthsub — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Pull request merged**